experiments/ClimaEarth/components/atmosphere/climaatmos_recursive.jl

```diff
@@ -0,0 +1,76 @@
+import ClimaComms
+import ClimaAtmos as CA
+import ClimaCore
+import ClimaCore: DataLayouts, Fields, Geometry
+import ClimaCore.Fields: Field, FieldVector, field_values
+import ClimaCore.DataLayouts: AbstractData
+import ClimaCore.Geometry: AxisTensor
+import ClimaCore.Spaces: AbstractSpace
+import ClimaUtilities.TimeVaryingInputs: AbstractTimeVaryingInput
+import NCDatasets
+
+# Allow cache to be moved to CPU (this is a little bit of type piracy...)
+ClimaCore.Adapt.@adapt_structure CA.AtmosCache
+ClimaCore.Adapt.@adapt_structure CA.RRTMGPInterface.RRTMGPModel
```

Comment:
We should not be doing type piracy like this. Please remove this.

Reply:
The reason I left them here is because this entire `atmosphere` module in ClimaEarth is intended to be something that ClimaAtmos should provide at some point (in an extension for ClimaCoupler, or in its source code). The same is true for the other modules in the `ClimaEarth/components` folder, and this is not the only type piracy that we have in there.

I am happy to move this directly to ClimaAtmos now.

Reply:
added to ClimaAtmos in https://github.com/CliMA/ClimaAtmos.jl/pull/3673